### PR TITLE
Simple Edge Walker

### PR DESCRIPTION
**Pull request by Bentleysb**
### Description:

This adds a simple implementation of the abstract EdgeWalker class. This saves creating a new implementation each time only a basic walker is needed. This walker allows the candidate selection and filtering functions to be passed in at instantiation. This makes it quite adaptable, as long the only output required is a single unsorted set of collected edges. 

### Potential Impact:

No immediate effect, as this is a new class. However, this can be used to replace existing custom walkers in some atlas-checks' checks (e.g InvalidLanesTagCheck, EdgeCrossingEdgeCheck, MalformedRoundaboutCheck)

### Unit Test Approach:

Unit test `collectAllTest` checks the `SimpleEdgeWalker(Edge, Function)` constructor, and checks the infinite loop protection built into the `collectEdges` method.
Unit test `collectPrimaryQueueAllTest` checks the `SimpleEdgeWalker(Edge, Predicate, Function)` constructor. 

### Test Results:

NA

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

**Review by nachtm — COMMENTED** [review]
Implementation looks solid, and I can definitely see this being helpful! I have two overall questions/comments:

- Should this go in atlas or atlas-checks? 
- I'm also wondering about your PR on ShadowDetection, which is also using a BFS. However, that's not searching over edges so we can't use this utility there. It would be super cool to generalize this further, though that's likely something to do down the road.

**Comment by Bentleysb** [comment]
I think this fits in atlas, as it is just a usability extension to the EdgeWalker class. If the EdgeWalker fits in Atlas, this does. 
I agree it would be great to have a BFS that could be used more generally. This is just a quick way to make an existing class more usable. We should definitely look into what that could look like and where that would live.

**Review by nachtm — APPROVED** [review]
Sounds good! Thanks for doing this :)

File: src/main/java/org/openstreetmap/atlas/geography/atlas/walker/SimpleEdgeWalker.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package org.openstreetmap.atlas.geography.atlas.walker;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+
+/**
+ * This class provides some constructors for simple {@link EdgeWalker}s.
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/walker/SimpleEdgeWalker.java]: Could you elaborate a bit more here?

**Review by danielduhh — CHANGES_REQUESTED** [review]
Had a single nit. Thanks for doing this; I wasn't fully convinced this belongs in Atlas, but you made a good point about the EdgeWalker being in the same package.

Also... nice name!

**Review by danielduhh — APPROVED** [review]

**Pull request closed without merging**